docs/sphinx_pyodide/sphinx_pyodide/__init__.py

```diff
@@ -12,7 +12,38 @@ def wrap_analyzer(app):
     app._sphinxjs_analyzer = PyodideAnalyzer(app._sphinxjs_analyzer)
 
 
+def patch_templates():
+    """Patch in a different jinja2 loader so we can override templates with our
+    own versions.
+    """
+    from pathlib import Path
+
+    from jinja2 import ChoiceLoader, Environment, FileSystemLoader, PackageLoader
+    from sphinx_js.analyzer_utils import dotted_path
+    from sphinx_js.renderers import JsRenderer
+
+    loader = ChoiceLoader(
+        [
+            FileSystemLoader(Path(__file__).parent / "templates"),
+            PackageLoader("sphinx_js", "templates"),
```

Comment:
By the way, it seems like [sphinx-js](https://github.com/mozilla/sphinx-js) is having some new contributors recently. So maybe we can upstream some of our changes.

Reply:
It seems like we have some significant difficulties in getting up to sphinx-js v3.2 -- sphinx-js pins typedoc version 0.15.0, but v0.15.0 cannot process our source files so we use v0.22.15. We also pin sphinx-js v3.1.0 and these work fine together. However, when you try to use sphinx-js v3.2.1 with typedoc 0.22.15, it crashes.

Hopefully upgrading to spinx-js 3.1.2 will be okay. Seems like we have to rewrite flatten-suffix-tree

Reply:
Thanks for the observation! So the life was not that easy :(